builtins/amp-ad.js

```diff
@@ -253,17 +253,20 @@ export function installAd(win) {
             this.element.creativeId = info.id;
           }, /* opt_is3P */ true);
           listen(this.iframe_, 'embed-size', data => {
+            let newHeight, newWidth;
             if (data.width !== undefined) {
+              newWidth = Math.max(this.element./*OK*/offsetWidth +
+                  data.width - this.iframe_./*OK*/offsetWidth, data.width);
               this.iframe_.width = data.width;
-              this.element.setAttribute('width', data.width);
+              this.element.setAttribute('width', newWidth);
             }
             if (data.height !== undefined) {
-              const newHeight = Math.max(this.element./*OK*/offsetHeight +
+              newHeight = Math.max(this.element./*OK*/offsetHeight +
                   data.height - this.iframe_./*OK*/offsetHeight, data.height);
               this.iframe_.height = data.height;
               this.element.setAttribute('height', newHeight);
-              this.updateHeight_(newHeight);
             }
+            this.updateSize_(newHeight, newWidth);
```

Comment:
`if (newHeight !== undefined || newWidth !== undefined) {this.ujpdateSize_(...)}`